lockrenewal.go

```diff
@@ -0,0 +1,77 @@
+package servicebus
+
+import (
+	"context"
+	"fmt"
+	"github.com/Azure/azure-amqp-common-go/log"
+	"github.com/Azure/azure-amqp-common-go/rpc"
+	otlogger "github.com/opentracing/opentracing-go/log"
+	"time"
+
+	"pack.ag/amqp"
+)
+
+const (
+	serviceBuslockRenewalOperationName = "com.microsoft:renew-lock"
+)
+
+//RenewLocks renews the locks on messages provided
+func (e *entity) RenewLocks(ctx context.Context, messages []*Message) error {
+	span, ctx := e.startSpanFromContext(ctx, "sb.entity.renewLocks")
+	defer span.Finish()
+
+	lockTokens := make([]amqp.UUID, 0, len(messages))
+	for _, m := range messages {
+		if m.LockToken == nil {
+			log.For(ctx).Error(fmt.Errorf("failed: message has nil lock token, cannot renew lock"), otlogger.Object("messageId", m))
+			continue
+		}
+
+		amqpLockToken := amqp.UUID(*m.LockToken)
+		lockTokens = append(lockTokens, amqpLockToken)
+	}
+
+	if len(lockTokens) < 1 {
+		log.For(ctx).Info("no lock tokens present to renew")
+		return nil
+	}
+
+	e.renewMessageLockMutex.Lock()
+	defer e.renewMessageLockMutex.Unlock()
```

Comment:
One benefit of moving this method to the message granularity is that we should be able to add a mutex at the message level instead of having a single instance of a lock renewer running at a time. Is there anything that you know of that challenges that assumption?